src/services/findAllReferences.ts

```diff
@@ -1337,6 +1337,30 @@ namespace ts.FindAllReferences {
             }
         }
 
+        export function getTopMostDeclarationNamesInFile(declarationName: string, sourceFile: SourceFile): readonly Node[] {
+            const candidates = filter(getPossibleSymbolReferenceNodes(sourceFile, declarationName), name => !!getDeclarationFromName(name));
+            return candidates.reduce((topMost, decl) => {
+                const depth = getDepth(decl);
+                if (!some(topMost.declarationNames) || depth === topMost.depth) {
+                    topMost.declarationNames.push(decl);
+                }
+                else if (depth < topMost.depth) {
+                    topMost.declarationNames = [decl];
+                }
+                topMost.depth = depth;

```

Comment:
I might be blanking, but it seems like you wouldn't want to do this if `depth > topMost.depth`?